README.md

```diff
@@ -50,3 +50,17 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+The worst-case asymptotic time complexity of my implementation would be $O(n^3)$. The reason for this is when breaking down the implementation, the random start function needs to generate a random route by shuffling an array of size n. This has a complexity of $O(n)$ due to the fact that it needs to iterate through the array in order to preform the swaps. Next the calculate route function computes the total distance of a given route. The route has a length of n so once the function iterates through the route, we receive a time complexity of $O(n)$. Next the required 2-opt swap function focuses on creating a new route by slicing and reversing pieces of the origional route. Both these operations have a complexity of $O(n)$. Finally within the main functionality of this program, the while loop continues as long as improvements can be made. The outer loop runs for each city so for $O(n)$ time. The inner loop also runs for this sae time, because it fruns for each possible swap. Each swap that happens, requires the creation of a new route and then calculating its distance, which also takes $O(n)$ time. Therefore, the nested loops have a time complexity of $O(n^3)$. Once we combine all the components this is where we get our final worst time complexity of $O(n^3)$. The memory complexity worst case would be different. It would be $O(n^2)$ because the input distance matrix is n*n, which requires $O(n^2)$ space. Then the route array and the temporary array in the 2-opt both rewuire $O(n)$ space. The functions themselves, such as the random start, route calculation, and the two opt itself don't use additional space other than from their constant memory used for the inputs and outputs. Once we combine these, we see that the worst memory complexity would be $O(n^2)$.
```

Comment:
How many times could you find an improving route, in the worst case?

Reply:
In the worst case, we would need to look at the possible swaps available I believe. So since in the worst case scenerio, every iteration finds at least one improving route before it ends, then the number would correlate to the number of iterations. So if every iteration only performs one improvement, then it we could find O(n^2) improving routes.

Reply:
How many different routes are there?

Reply:
I believe there are n! routes which based off of the number of cities and their possible arrangements

Reply:
Ok, and what does that tell you about the number of times an improving route could be found?

Reply:
As the number of cities increases, the number of route found would become exceptionally bigger. Making it very difficult to find an improving route since there are so many routes to consider. Isn't this one of the main reasons that held-karp and local search are different? Isn't it because you are guaranteed to find the shortest path since the search space is smaller in held-karp, whereas in this local search I guess we are not guaranteed to find the shortest path since we used decently small test cases? I know that is a question on top of an answer, but that is my understanding of the two at the moment as i've worked through them.

Reply:
Yes, but the question here is what the asymptotic complexity of your implementation is. And that depends primarily on the number of times an improving route can be found.

Reply:
Okay so after working through this with you I know that I need to include the n!, so would my final complexity be O(n! * n^3)? I still need to acknowledge the complexty for the rest of the iterations of my problem while adding in the complexity for the routes.

Reply:
That sounds like it's going in the right direction.

Reply:
I will make those changes to my readme, thank you for working with me on this problem!